File: dotnet/Lexy/Lexy.Compiler/Language/TypeNames.cs
using System.Collections.Generic;

namespace Lexy.Compiler.Language
{
    public static class TypeNames
    {
        private static readonly IList<string> existing = new List<string>
        {
            Number,
            Boolean,
            Date,
            String
        };

        public const string Number = "number";
        public const string Boolean = "boolean";
        public const string Date = "date";
        public const string String = "string";

        public static bool Contains(string parameterType)
        {
            return existing.Contains(parameterType);
        }
    }
}

File: typescipt/src/language/tables/typeNames.ts


export class TypeNames {
   public const string Number = `number`;
   public const string Boolean = `boolean`;
   public const string Date = `date`;
   public const string String = `string`;

   private static readonly Array<string> existing = new Array<string> {
     Number,
     Boolean,
     Date,
     String
   };

   public static contains(parameterType: string): boolean {
     return existing.contains(parameterType);
   }
}
